src/target/riscv/riscv-013.c

```diff
@@ -831,25 +828,31 @@ static uint32_t __attribute__((unused)) abstract_register_size(unsigned width)
 
 static int wait_for_idle(struct target *target, uint32_t *abstractcs)
 {
-	RISCV013_INFO(info);
+	assert(target);
+	assert(abstractcs);
+
 	time_t start = time(NULL);
-	while (1) {
-		if (dm_read(target, abstractcs, DM_ABSTRACTCS) != ERROR_OK)
+	do {
+		if (dm_read(target, abstractcs, DM_ABSTRACTCS) != ERROR_OK) {
+			/* NOTE: this value is meaningless, overwrite to avoid stale value */
```

Comment:
Did I please get correctly the meaning of this comment?

```suggestion
			/* We couldn't read abstractcs. For safety, overwrite the output value to prevent the caller
			 * working with a stale value of abstractcs. */
```

Reply:
yep